components/Dialogs/tests/snapshot/MDCAlertControllerDynamicTypeTests.m

```diff
@@ -0,0 +1,308 @@
+  // Given
+  [self setAlertControllerContentSizeCategory:UIContentSizeCategoryAccessibilityExtraLarge];
+  self.alertController.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = NO;
+
+  // When
+  self.alertController.mdc_adjustsFontForContentSizeCategory = YES;
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryAccessibilityExtraExtraLarge {
+  // Given
+  [self setAlertControllerContentSizeCategory:
+            UIContentSizeCategoryAccessibilityExtraExtraLarge];
+  self.alertController.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = NO;
+
+  // When
+  self.alertController.mdc_adjustsFontForContentSizeCategory = YES;
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryAccessibilityExtraExtraExtraLarge {
+  // Given
+  [self setAlertControllerContentSizeCategory:
+            UIContentSizeCategoryAccessibilityExtraExtraExtraLarge];
+  self.alertController.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = NO;
+
+  // When
+  self.alertController.mdc_adjustsFontForContentSizeCategory = YES;
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)testScaledFontDynamicTypeForContentSizeCategoryExtraSmallAndLegacyEnabled {
+  // Given
+  [self setAlertControllerContentSizeCategory:UIContentSizeCategoryExtraSmall];
+  self.alertController.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = YES;
+
+  // When
+  self.alertController.mdc_adjustsFontForContentSizeCategory = YES;
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
+}
+
+- (void)
+    testScaledFontDynamicTypeForContentSizeCategoryAccessibilityExtraExtraExtraLargeAndLegacyEnabled {
+  // Given
+  [self setAlertControllerContentSizeCategory:
+            UIContentSizeCategoryAccessibilityExtraExtraExtraLarge];
+  self.alertController.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = YES;
+
+  // When
+  self.alertController.mdc_adjustsFontForContentSizeCategory = YES;
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.alertController.view];
```

Comment:
Why didn't the fonts change in this case? I'd have expected the legacy behavior to result in the font sizes changing here.

Reply:
I believe the legacy doesn't scale in this use case because we aren't mocking UIApplication or UIScreen instead we are just faking the perferredContentSize for the component. The legacy behavior inspects the UIApplication and UIScreen.

https://github.com/material-components/material-components-ios/blob/1df302cd48b4384cbb27718070f6a84799cb24bd/components/Typography/src/UIFontDescriptor%2BMaterialTypography.m#L68-L81

Reply:
Ah. That's a mega bummer.

Reply:
Can you add a comment to the tests indicating as such so that others in the future aren't surprised?